core/bin/zksync_server/src/main.rs

```diff
@@ -175,9 +175,9 @@ fn main() -> anyhow::Result<()> {
 
     let result = node.run(observability_guard);
     if find_gateway_update_error(&result) {
-        tracing::warn!("Gateway has changed")
+        tracing::warn!("Settlement layer has changed")
     };
-    Ok(())
+    Ok(result?)
```

Comment:
O think you can just return `result`

Reply:
we need conversion from `ZkStackServiceError` to anyhow::Error